### PR TITLE
Commits over time chart

### PR DESCRIPTION
**Pull request by herfstregen**

### CODE REVIEW

File: src/main/resources/templates/project-commit-graph.ftl [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+[@macros.renderMenu i18n user /]
+<div class="container">
+
+[#if repositoryEntity?? && repositoryEntity?has_content]
+    [@projectFrameset.renderBreadcrumb i18n group![] repositoryEntity/]
+[/#if]
+
+    <div class="row">
+        <div class="col-md-10 col-md-offset-2">
+            <h4 style="line-height:34px; margin-top:0;">Nice graphs &nbsp; 😍</h4>
```
Comment (JWGmeligMeyling) [inline src/main/resources/templates/project-commit-graph.ftl]: Can you please replace all "Nice graphs" occurrences with a property within our localisation file? (so `${i18n.translate("header.graphs")}` and then in `devhub_en.properties` you add: `header.graphs = ...`. We may reconsider to change the title too. I would keep the emoji.

File: src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java [hunk 2]
```diff
@@ -780,4 +797,76 @@ public void enhanceCommitsForRepository() {
...
+	@Produces(MediaType.APPLICATION_JSON)
+
+	public List<List<Object>> getMagicalChartData() {
+
+		List<Commit> commitEntities = getAllCommits();
+
+		return getGoogleChartDataForCommits(commitEntities);
+	}
+
+	private static List<List<Object>> getGoogleChartDataForCommits(List<Commit> commitEntities) {
```
Comment (TimvdLippe) [inline src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java]: I don't see any mention or communication with Google in this method. Should it be called `getGoogleChartDataForCommits`?

File: src/main/resources/templates/project-commit-graph.ftl [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                            title: name,
+                            colors: [randomcolor],
+                            pointShape: 'diamond',
+                            pointSize: 5,
+                            backgroundColor: 'transparent',
+                            hAxis: {minValue: minDate, maxValue: maxDate, title: 'Date',  titleTextStyle: {color: '#333'}, gridlines: {color : 'transparent'}},
+                            vAxis: {minValue: 0, gridlines: {count : -1}}
+                        };
+
+                        var mijnMagischeElement = $('<div>')
```
Comment (TimvdLippe) [inline src/main/resources/templates/project-commit-graph.ftl]: Dutch?

File: src/main/resources/templates/project-commit-graph.ftl [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+                var chart = new google.visualization.AreaChart(document.getElementById('allcommits_div'));
+                chart.draw(data, options);
+            });
+    });
+
+
+    google.charts.load('current', {'packages':['corechart']});
+    google.charts.setOnLoadCallback(function() {
+        $.get('http://localhost:50001/courses/ti1705/TI1705/groups/1/person-commit')
```
Comment (TimvdLippe) [inline src/main/resources/templates/project-commit-graph.ftl]: Should be the production url, or relative from the current url.

File: src/main/resources/templates/project-commit-graph.ftl [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        var rgb = [Math.random() * 256, Math.random() * 256, Math.random() * 256];
+        var mix = [brightness*51, brightness*51, brightness*51]; //51 => 255/5
+        var mixedrgb = [rgb[0] + mix[0], rgb[1] + mix[1], rgb[2] + mix[2]].map(function(x){ return Math.round(x/2.0)})
+        return "rgb(" + mixedrgb.join(",") + ")";
+    }
+
+    var randomcolor = getRandColor(4);
+    google.charts.load('current', {'packages':['corechart']});
+    google.charts.setOnLoadCallback(function() {
+        $.get('http://localhost:50001/courses/ti1705/TI1705/groups/1/magical-chart-data')
```
Comment (TimvdLippe) [inline src/main/resources/templates/project-commit-graph.ftl]: Should be the production url, or relative from the current url.

File: src/main/resources/templates/project-commit-graph.ftl [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    </div>
+
+</div> <!-- closes div class="container" -->
+
+[@macros.renderScripts /]
+<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+<script type="text/javascript" src="http://code.stephenmorley.org/javascript/colour-handling-and-processing/Colour.js"></script>
+<script type="text/javascript">
+
+    function getRandColor(brightness){
```
Comment (TimvdLippe) [inline src/main/resources/templates/project-commit-graph.ftl]: Did you copy this from an external resource? Would be better to include a link to it, maybe StackOverflow?

File: src/main/resources/templates/project-commit-graph.ftl [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    </div>
+
+    <div class="row">
+        <div class="col-md-2">
+        [@projectFrameset.renderSidemenu "insights" i18n group![] repository/]
+        </div>
+
+        <!-- Graph for all commits -->
+        <div class="col-md-10">
+            <div id="allcommits_div" style="width: 100%; height: 500px; "></div>
```
Comment (TimvdLippe) [inline src/main/resources/templates/project-commit-graph.ftl]: Include styling in a `<style>` instead, not in-line

File: src/main/resources/templates/project-commit-graph.ftl (hunk 6 above)
Comment (asylunatic) [inline src/main/resources/templates/project-commit-graph.ftl]: I think I did yes, you want a link for reference?

**Comment by TimvdLippe** [comment]
Yes, because it is complex code and not immediately clear how it works.

File: src/main/resources/templates/project-commit-graph.ftl (hunk 7 above)
Comment (Fastjur) [inline src/main/resources/templates/project-commit-graph.ftl]: Or in the `sass` files perhaps.

File: src/main/resources/templates/project-commit-graph.ftl (hunk 5 above)
Comment (JWGmeligMeyling) [inline src/main/resources/templates/project-commit-graph.ftl]: It should be relative indeed. (`/courses...`)

File: src/main/resources/templates/project-commit-graph.ftl (hunk 4 above)
Comment (JWGmeligMeyling) [inline src/main/resources/templates/project-commit-graph.ftl]: It should be relative indeed. (`/courses...`)

File: src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java (hunk 2 above)
Comment (JWGmeligMeyling) [inline src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java]: It doesn't call Google but it represents a data format specifically for Google Charts. So the naming is fine.

File: src/main/resources/templates/project-commit-graph.ftl (hunk 5 above)
Comment (Fastjur) [inline src/main/resources/templates/project-commit-graph.ftl]: And it should dynamically get the group and course numbers

Comment (JWGmeligMeyling) [inline src/main/resources/templates/project-commit-graph.ftl]: Good catch :wink:

File: src/main/resources/templates/project-commit-graph.ftl (hunk 1 above)
Comment (asylunatic) [inline src/main/resources/templates/project-commit-graph.ftl]: I did, but it killed the emoji :'(

File: src/main/resources/templates/project-commit-graph.ftl (hunk 3 above)
Comment (asylunatic) [inline src/main/resources/templates/project-commit-graph.ftl]: ¯\_(ツ)_/¯ maybe to make the magic even more magic? :')

File: src/main/resources/templates/project-commit-graph.ftl [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+
+                var chart = new google.visualization.AreaChart(document.getElementById('allcommits_div'));
+                chart.draw(data, options);
+            });
+    });
+
+
+    google.charts.load('current', {'packages':['corechart']});
+    google.charts.setOnLoadCallback(function() {
+        $.get('/courses/ti1705/TI1705/groups/1/person-commit')
```
Comment (asylunatic) [inline src/main/resources/templates/project-commit-graph.ftl]: Also this should dynamically load, haven't figured that out though

**Pull request merged**